### PR TITLE
Amls 4347

### PR DESCRIPTION
**Pull request by slavvco**
Changes to sole proprietor question: 
- changed flow to not display sole proprietor question when people will update date of birth or fit and proper data but to redirect them appropriately  to VAT or SA page

## Related / Dependant PRs?

None

## How Has This Been Tested?

Unit tests, manual tests

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.

### CODE REVIEW

File: test/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessControllerSpec.scala [hunk 1]
```diff
@@ -19,16 +19,16 @@ package controllers.responsiblepeople
 import connectors.DataCacheConnector
```
Comment (drewmiley) [inline test/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessControllerSpec.scala]: Is the indentation right in this file?

File: app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala [hunk 2]
```diff
@@ -74,4 +84,12 @@ class SoleProprietorOfAnotherBusinessController @Inject()(val dataCacheConnector
         }
       }
   }
+
+  def getViewForVat(vatReg: Option[VATRegistered], index: Int, edit: Boolean = false, flow: Option[String] = None) = {
```
Comment (drewmiley) [inline app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala]: Refactor into case statement?

File: app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala [hunk 3]
```diff
@@ -30,17 +29,28 @@ import scala.concurrent.Future
...
-          => Ok(sole_proprietor(EmptyForm, edit, index, flow, personName.titleName))
-          case _ => NotFound(notFoundView)
+
+
+      implicit authContext => implicit request => {
+        for {
+          responsiblePerson <- getData[ResponsiblePerson](index)
+        } yield {
+          responsiblePerson match {
+            case Some(ResponsiblePerson(Some(personName), _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, Some(soleProprietorOfAnotherBusiness)))
```
Comment (drewmiley) [inline app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala]: Refactor out this?

File: app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala [hunk 4]
```diff
@@ -30,17 +29,28 @@ import scala.concurrent.Future
...
+            => Ok(sole_proprietor(Form2[SoleProprietorOfAnotherBusiness](soleProprietorOfAnotherBusiness), edit, index, flow, personName.titleName))
+            case Some(ResponsiblePerson(Some(personName), _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, None, _, _, _))
```
Comment (drewmiley) [inline app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala]: What is the None, just for my understanding?

File: app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala (hunk 2 above)
Comment (slavvco) [reply app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala]: done

File: test/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessControllerSpec.scala (hunk 1 above)
Comment (slavvco) [reply test/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessControllerSpec.scala]: I hope it is so

Comment (drewmiley) [reply test/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessControllerSpec.scala]: Ok :)

File: app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala (hunk 4 above)
Comment (drewmiley) [reply app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala]: Ok :)

File: app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala (hunk 2 above)
Comment (drewmiley) [reply app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala]: Ah i meant to refactor it into the earlier case statement?

Comment (slavvco) [reply app/controllers/responsiblepeople/SoleProprietorOfAnotherBusinessController.scala]: You need to be more explicit with me:P

Done

**Pull request merged**